### ISSUE TITLE
tf.nn.silu returns wrong gradient for large negative float64 input

### ISSUE THREAD

**Issue by ALinrunrun** [tensorflow/tensorflow#124835]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

tf 2.21.0

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 22.04

### Python version

Python 3.13

### Current behavior?

`tf.nn.silu` returns an incorrect gradient for a large negative float64 input.

For SiLU / swish, the function is `x * sigmoid(x)`. At `x = -709.0`, the analytic high-precision reference for the derivative is approximately `-8.614807714413836e-306`, which is a finite nonzero float64 value.

However, TensorFlow autodiff returns `-0.0` for this gradient. PyTorch computes the expected finite value for the same derivative, so this appears to be a backward-pass issue rather than a float64 limitation.

Expected behavior?

The gradient of `tf.nn.silu` at `x = -709.0` should be close to the analytic reference value `-8.614807714413836e-306`, not `-0.0`.

### Standalone code to reproduce the issue

```shell
import os
os.environ.setdefault("CUDA_VISIBLE_DEVICES", "-1")
os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "3")

import tensorflow as tf

x = -709.0
truth = -8.614807714413836e-306
rtol = 1e-6

v = tf.Variable(x, dtype=tf.float64)

with tf.GradientTape() as tape:
    y = tf.nn.silu(v)

g = tape.gradient(y, v)

actual = float(g)
rel_err = abs(actual - truth) / abs(truth)

print(f"x: {x!r}")
print(f"tensorflow gradient: {actual!r}")
print(f"expected gradient: {truth!r}")
print(f"relative error: {rel_err:.3e}")

if rel_err > rtol:
    print("BUG REPRODUCED: tf.nn.silu gradient is wrong for large negative float64 input")
else:
    print("not reproduced")
```

### Relevant log output

```shell
x: -709.0
tensorflow gradient: -0.0
expected gradient: -8.614807714413836e-306
relative error: 1.000e+00
BUG REPRODUCED: tf.nn.silu gradient is wrong for large negative float64 input
```

**Comment by GodlyDonuts** [comment]
I reproduced this on TensorFlow 2.20.0 and am working on a focused fix with regression coverage. The custom SiLU gradient recomputes sigmoid(beta * x), which underflows to zero for a large negative float64 input even though the analytic derivative remains representable.
